### PR TITLE
Roll NPM packages and Node.js

### PR DESCRIPTION
**Pull request by nekketsuuu**
- Remove sass
- Upgrade Node.js to v20 (the current LTS)
- Upgrade Webpack-related packages to the latest, font-awesome to the latest, AdminLTE to v3, Bootstrap to v4, and jQuery to v3

Please refer to each commit for details.

## Screenshots

| Before | After |
| -- | -- |
|  |  |
|  |  |

### CODE REVIEW

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
-import 'admin-lte/bootstrap/css/bootstrap.css';
```
Comment (nekketsuuu) [inline app/javascript/packs/application.js:1]: Note: Importing bootstrap.css here is unnecessary, and also it sometimes leads to double loading errors. So I just removed this line.

- AdminLTE itself loads its bundled bootstrap.css automatically, as https://adminlte.io/docs/3.2/upgrade-guide.html says "You will not need to load the Bootstrap CSS file bootstrap(.min).css, but you will need the Bootstrap JS file bootstrap(.min).js."
- One of AdminLTE dependencies, bootstrap-switch, depends on bootstrap v3 (not v4). Hence node_modules/bootstrap is v3, and node_modules/admin-lte/node_modules/bootstrap is v4.
- I haven't checked out why, but importing node_modules/admin-lte/node_modules/bootstrap/dist/css/bootstrap.css manually causes conflicts of style.

**Comment by mozamimy** [comment]
The looks of list items in 'Database' (e.g. localhost_dmemo_development) in the left sidebar you showed 'After' screenshots seem quite changed, is that not a problem? 🤔

File: app/javascript/dmemo.css [hunk 2]
```diff
@@ -0,0 +1,127 @@
+.skin-blue {
+  a {
```
Comment (os0x) [inline app/javascript/dmemo.css:2]: https://caniuse.com/css-nesting is available 👍

File: app/views/data_sources/_form.html.haml [hunk 3]
```diff
@@ -1,55 +1,55 @@
...
 
-  .box-footer
-    = f.submit class: "btn btn-primary pull-right"
+  .card-footer
+    = f.submit class: "btn btn-primary float-right"
 
 :javascript
-  $(function() {
+  $(() => {
     var standardAdpaters = #{DataSourceAdapters.standard_adapter_names.to_json.html_safe};
```
Comment (os0x) [inline app/views/data_sources/_form.html.haml:53]: [nits] use https://api.rubyonrails.org/classes/ActionView/Helpers/JavaScriptHelper.html#method-i-escape_javascript

Comment (os0x) [reply app/views/data_sources/_form.html.haml:53]: otherwise data attribute is better

**Comment by nekketsuuu** [comment]
https://github.com/cookpad/dmemo/pull/332#issuecomment-1979911715

This is intentional. Since I made the sidebar collapsible when clicking the upper left `.fa-bars` icon, each menu item now requires its icon on the left side. The following screenshot shows how icons are rendered when the sidebar is collapsed. Then the previous design does not fit well with this collapsed sidebar.



Without the collapsible sidebar, we can't access certain pages, such as `/users`, on a narrow view (e.g., on mobile). This is because why I want to change the sidebar to be collapsible.

Do you prefer the previous design? I'd appreciate any suggestions for improving the current design. I'm open to modifying the implementation based on feedback.

**Comment by mozamimy** [comment]
I got about the intention of a change you made! I have preferred the collapsable sidebar after understanding your thinking.

**Review by mozamimy — APPROVED** [review]

**Pull request merged**